src/hooks/use-hackathon-dates.ts

```diff
@@ -0,0 +1,37 @@
+import { getHackathonDates } from "@/services/firebase/hackathon";
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+import { useCallback } from "react";
+
+/**
+ * Hook for accessing hackathon dates data
+ */
+
+const CURRENT_HACKATHON_YEAR = "2025";
+
+export const useHackathonDates = () => {
+	const queryClient = useQueryClient();
+
+	const queryKey = ["hackathons", CURRENT_HACKATHON_YEAR];
+	const { data: hackathonDates = null, isLoading } = useQuery({
+		queryKey,
+		queryFn: async () => {
+			return await getHackathonDates(CURRENT_HACKATHON_YEAR);
+		},
+		enabled: !!CURRENT_HACKATHON_YEAR,
+		refetchOnWindowFocus: false,
+		staleTime: Number.POSITIVE_INFINITY, // application data only really changes after a year so there is no need to refetch 
+	});
+
+	/**
+	 * Invalidates and refreshes the hackathon dates query
+	 */
+	const refreshHackathonDates = useCallback(() => {
+		return queryClient.invalidateQueries({ queryKey });
+	}, [queryClient, queryKey]);
+
+	return {
+		hackathonDates: hackathonDates, 
+		refreshHackathonDates,
+		isLoading,
+	};
+};
```

Comment:
Couple things:
- I think the whole point of fetching the application start date and end date from firebase rather than hard-coding them in the React app is that the user doesn't need to do a hard refresh when we update the dates, it should just magically update in their app
  - With that in mind, we don't want to set the staleTime to infinity and want to consider refetching on window focus
- `!!CURRENT_HACKATHON_YEAR` is always going to evaluate to true
- We are not using `refreshHackathonDates` anywhere so I would remove it... but if we do need it at some point `useQuery` returns `refetch` which does the same thing as `refreshHackathonDates`

```suggestion
export const useHackathonDates = () => {
	const { data: hackathonDates = null, isLoading } = useQuery({
		queryKey: ["hackathons", CURRENT_HACKATHON_YEAR],
		queryFn: () => getHackathonDates(CURRENT_HACKATHON_YEAR),
	});

	return {
		hackathonDates: hackathonDates,
		isLoading,
	};
};
```